algorithms/HPdclassifier/R/hpdrpart.R

```diff
@@ -0,0 +1,279 @@
+#####################################################################################
+# Copyright [2013] Hewlett-Packard Development Company, L.P.                        # 
+#                                                                                   #
+# This program is free software; you can redistribute it and/or                     #
+# modify it under the terms of the GNU General Public License                       #
+# as published by the Free Software Foundation; either version 2                    #
+# of the License, or (at your option) any later version.                            #
+#                                                                                   #
+# This program is distributed in the hope that it will be useful,                   #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of                    #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                      #
+# GNU General Public License for more details.                                      #
+#                                                                                   #
+# You should have received a copy of the GNU General Public License                 #
+# along with this program; if not, write to the Free Software                       #
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.    #
+#####################################################################################
+
+
+hpdrpart <- function(formula, data, weights, subset , na.action = na.omit, 
+	 model = FALSE, x = FALSE, y = FALSE, params = NULL, 
+	 control = NULL, cost = NULL, 
+	 completeModel = TRUE, nBins = 256L, do.trace = FALSE)
+{
+
+	if(!identical(na.action, na.exclude) &
+		!identical(na.action, na.omit) &
+		!identical(na.action, na.fail))
+		stop("'na.action' must be either na.exclude, na.omit, na.fail")
+
+	if(!missing(subset))
+		warning("'subset' not implemented. Adjust using weights parameter")
+	if(missing(subset))
+		subset = NULL
+	if(is.null(control))
+		control = rpart.control()
+
+	if(missing(formula))
+		stop("'formula' is a required argument")
+	if(!inherits(formula, "formula"))
+		stop("'formula' is not of class formula")
+
+	if(missing(data))
+		stop("'data' is a required argument")
+	if(!is.dframe(data) & !is.data.frame(data))
+		stop("'data' must be a dframe or data.frame")
+	if(is.data.frame(data))
```

Comment:
Please add validation check for weights. I think the number of its rows and its type should be the same as the number of rows of data. Moreover, they should have the same partition schema whenever both are dframe. You can check it with comparing their result of partitionsize().